src/FixedRankMatrices.jl

```diff
@@ -0,0 +1,177 @@
+
+# Constructors
+* `UMVTVector(U,S,Vt)` store umv factors to initialize the `UMVTVector`
+* `UMVTVector(U,S,Vt,k)` store the umv factors after shortening them down to
+  inner dimensions $k$, i.e. in $UMV^\mathrm{T}$, $M\in\mathbb R^{k\times k}$
+"""
+struct UMVTVector{T} <: TVector
+    U::Matrix{T}
+    M::Matrix{T}
+    Vt::Matrix{T}
+end
+UMVTVector(U,M,Vt,k::Int) = UMVTVector(U[:,1:k],M[1:k,1:k],Vt[1:k,:])
+UMVTVector(U,M,Vt) = UMVTVector{eltype(U)}(U,M,Vt)
+
+function check_manifold_point(F::FixedRankMatrices{M,N,k,T},x; kwargs...) where {M,N,k,T}
+    r = rank(x; kwargs...)
+    s = "The point $(x) does not lie on the manifold of fixed rank matrices of size ($(M),$(N)) witk rank $(k), "
+    if size(x) != (M,N)
+        return DomainError(size(x), string(s,"since its size is wrong."))
+    end  
+    if r > k
+        return DomainError(r, string(s, "since its rank is too large."))
+    end 
+    return check_manifold_point(F,SVDMPoint(x,k))
+end
+
+function check_manifold_point(F::FixedRankMatrices{M,N,k,T}, x::SVDMPoint) where {M,N,k,T}
+    s = "The point $(x) does not lie on the manifold of fixed rank matrices of size ($(M),$(N)) witk rank $(k), "
+    if (size(x.U) != (M,k)) || (length(x.S) != k) || (size(x.Vt) != (k,N))
+        return DomainError([size(x.U)...,length(x.S),size(x.Vt)], string(s, "since the dimensions do not fit (expected $([N,k,k,k,M]))."))
+    end
+    if !isapprox(x.U'*x.U,one(zeros(M,M)))
+        return DomainError(norm(x.U'*x.U-one(zeros(M,M))), string(s," since U is not orthonormal/unitary."))
+    end
+    if !isapprox(x.Vt'*x.Vt, one(zeros(N,N)))
+        return DomainError(norm(x.Vt'*x.Vt-one(zeros(N,N))), string(s," since V is not orthonormal/unitary."))
+    end
+end
+
+function check_tangent_vector(F::FixedRankMatrices{M,N,k,T}, x::SVDMPoint, v::UMVTVector) where {M,N,k,T}
+    c = check_manifold_point(x)
+    if c != nothing
+        return c
+    end
+    if (size(v.U) != (M,k)) || (size(v.Vt) != (k,N)) || (size(v.M) != (k,k))
+        return DomainError(cat(size(v.U),size(v.M),size(v.Vt),dims=1), "The tangent vector $(v) is not a tangent vector to $(x) on the fixed rank matrices since the matrix dimensions to not fit (expected $([M,k,k,k,k,N])).")
+    end
+    if !isapprox(v.U'*x.U, zeros(k,k))
+        return DomainError(norm(v.U'*x.U-zeros(k,k)), "The tangent vector $(v) is not a tangent vector to $(x) on the fixed rank matrices since v.U'x.U is not zero. ")
+    end
+    if !isapprox(v.Vt*x.Vt', zeros(k,k))
+        return DomainError(norm(v.Vt*x.Vt-zeros(k,k)), "The tangent vector $(v) is not a tangent vector to $(x) on the fixed rank matrices since v.V'x.V is not zero.")
+    end
+end
+
+function inner(::FixedRankMatrices{M,N,k,T}, x::SVDMPoint, v::UMVTVector, w::UMVTVector) where {M,N,k,T}
+    return dot(v.U,w.U) + dot(v.M,w.M) + dot(v.Vt,w.Vt)
+end
+
+manifold_dimension(::FixedRankMatrices{M,N,k,T}) where {M,N,k,T <: Real} = (M+N-k)*k
```

Comment:
If you change the constructor for `FixedRankMatrices` as suggested, this would just be
```julia
manifold_dimension(::FixedRankMatrices{M,N,k,Real}) where {M,N,k} = (M+N-k)*k
manifold_dimension(::FixedRankMatrices{M,N,k,Complex}) where {M,N,k} = 2*(M+N-k)*k
```